experiments/tf_trainer/tf_rnn_model.py

```diff
@@ -0,0 +1,112 @@
+"""Tensorflow Estimator implementation of RNN Model with Attention"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+
+
+class TFRNNModel():
+  DEFAULT_HPARAMS = tf.contrib.training.HParams(max_seq_length=300)
+
+  def __init__(self, text_feature_name, target_labels):
+    self._text_feature_name = text_feature_name
+    self._target_labels = target_labels
+
+  def estimator(self, run_config):
+    estimator = tf.estimator.Estimator(
+        model_fn=self._model_fn,
+        params=TFRNNModel.DEFAULT_HPARAMS,
+        config=run_config)
+    return estimator
+
+  def _model_fn(self, features, labels, mode, params, config):
+    inputs = features[self._text_feature_name]
+    batch_size = tf.shape(inputs)[0]
+
+    rnn_layers = [
+        tf.nn.rnn_cell.GRUCell(num_units=size, activation=tf.nn.tanh)
+        for size in [128]
+    ]
+
+    # create a RNN cell composed sequentially of a number of RNNCells
+    multi_rnn_cell = tf.nn.rnn_cell.MultiRNNCell(rnn_layers)
+
+    outputs, states = tf.nn.dynamic_rnn(
+        multi_rnn_cell,
+        inputs,
+        sequence_length=tf.fill(dims=[batch_size], value=params.max_seq_length),
+        dtype=tf.float32)
+    outputs = tf.reshape(outputs, [batch_size, params.max_seq_length, 128])
+
+    unstacked_outputs = tf.unstack(outputs, num=params.max_seq_length, axis=1)
+
+    attention = tf.expand_dims(
+        tf.nn.softmax(
+            tf.concat(
+                [
+                    tf.layers.dense(
+                        inputs=tf.layers.dense(
+                            inputs=output, units=64, activation=tf.nn.relu),
+                        units=1,
+                        activation=None) for output in unstacked_outputs
+                ],
+                axis=1),
+            axis=-1), -1)
+
+    weighted_output = tf.multiply(attention, outputs)
+    weighted_output = tf.reduce_sum(weighted_output, -2)
+
+    logits = tf.layers.dense(
```

Comment:
Perhaps we can make this a hyperparameter as well (final_fully_connected_layer_sizes)